File: tests/benchmarks/ruby/word_count.rb
#!/usr/bin/env ruby

require './utils.rb'

def word_count(str)
  freqs = Hash.new 0
  str.split.each do |w|
    freqs[w.downcase] += 1
  end
  freqs
end

ARGV.each do |f|
  t = benchmark { with_utf8_file(f) { |c| word_count(c) } }
  STDERR.puts "#{f}: #{t}"
end

File: tests/benchmarks/python/word_count.py
#!/usr/bin/env python

import utils, sys

def word_count(string):
    freqs = {}
    for w in string.split():
        w = w.lower()
        if freqs.get(w):
            freqs[w] += 1
        else:
            freqs[w] = 1
    return freqs

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: utils.with_utf8_file(f, word_count))
    sys.stderr.write('{0}: {1}\n'.format(f, t))
